### PR TITLE
network: split-out sd_ipv4acd object management from Address object

### PR DESCRIPTION
**Pull request by yuwata**
Split-out of #28325.

### CODE REVIEW

File: src/network/networkd-ipv4acd.c [hunk 1]
```diff
@@ -129,6 +127,12 @@ static void on_acd(sd_ipv4acd *acd, int event, void *userdata) {
 
         link = address->link;
 
+        r = sd_ipv4acd_get_address(acd, &a);
```
Comment (bluca) [inline src/network/networkd-ipv4acd.c:133]: this change does not seem related to the commit? it only says `network/ipv4acd: use IN4_ADDR_TO_STRING()`

Comment (yuwata) [reply src/network/networkd-ipv4acd.c:133]: Indeed. I mixed change to the last commit. Now it is moved.

**Review by bluca — APPROVED** [review]
